File: projects/tox4j/src/test/java/im/tox/tox4j/core/exceptions/ToxFriendCustomPacketExceptionTest.java
package im.tox.tox4j.core.exceptions;

import im.tox.tox4j.ToxCoreTestBase;
import im.tox.tox4j.core.ToxCore;
import org.junit.Test;
import scala.runtime.BoxedUnit;

import static org.junit.Assert.assertEquals;

public class ToxFriendCustomPacketExceptionTest extends ToxCoreTestBase {

  @Test
  public void testSendLossyPacketNotConnected() throws Exception {
    try (ToxCore<BoxedUnit> tox = newTox()) {
      int friendNumber = addFriends(tox, 1);
      try {
        tox.sendLossyPacket(friendNumber, new byte[]{(byte) 200, 0, 1, 2, 3});
        fail();
      } catch (ToxFriendCustomPacketException e) {
        assertEquals(ToxFriendCustomPacketException.Code.FRIEND_NOT_CONNECTED, e.code());
      }
    }
  }

  @Test
  public void testSendLosslessPacketNotConnected() throws Exception {
    try (ToxCore<BoxedUnit> tox = newTox()) {
      int friendNumber = addFriends(tox, 1);
      try {
        tox.sendLosslessPacket(friendNumber, new byte[]{(byte) 160, 0, 1, 2, 3});
        fail();
      } catch (ToxFriendCustomPacketException e) {
        assertEquals(ToxFriendCustomPacketException.Code.FRIEND_NOT_CONNECTED, e.code());
      }
    }
  }

  @Test
  public void testSendLossyPacketNotFound() throws Exception {
    try (ToxCore<BoxedUnit> tox = newTox()) {
      try {
        tox.sendLossyPacket(0, new byte[]{(byte) 200, 0, 1, 2, 3});
        fail();
      } catch (ToxFriendCustomPacketException e) {
        assertEquals(ToxFriendCustomPacketException.Code.FRIEND_NOT_FOUND, e.code());
      }
    }
  }

  @Test
  public void testSendLosslessPacketNotFound() throws Exception {
    try (ToxCore<BoxedUnit> tox = newTox()) {
      try {
        tox.sendLosslessPacket(0, new byte[]{(byte) 160, 0, 1, 2, 3});
        fail();
      } catch (ToxFriendCustomPacketException e) {
        assertEquals(ToxFriendCustomPacketException.Code.FRIEND_NOT_FOUND, e.code());
      }
    }
  }

}

File: projects/tox4j/src/test/java/im/tox/tox4j/core/exceptions/ToxFriendCustomPacketExceptionTest.scala
package im.tox.tox4j.core.exceptions

import im.tox.tox4j.core.ToxCoreConstants
import im.tox.tox4j.testing.ToxTestMixin
import org.scalatest.FunSuite

final class ToxFriendCustomPacketExceptionTest extends FunSuite with ToxTestMixin {

  test("SendLosslessPacketNotConnected") {
    interceptWithTox(ToxFriendCustomPacketException.Code.FRIEND_NOT_CONNECTED)(
      _.friendSendLosslessPacket(0, Array[Byte](160.toByte, 0, 1, 2, 3))
    )
  }

  test("SendLossyPacketNotConnected") {
    interceptWithTox(ToxFriendCustomPacketException.Code.FRIEND_NOT_CONNECTED)(
      _.friendSendLossyPacket(0, 200.toByte +: Array.ofDim[Byte](4))
    )
  }

  test("SendLosslessPacketNotFound") {
    interceptWithTox(ToxFriendCustomPacketException.Code.FRIEND_NOT_FOUND)(
      _.friendSendLosslessPacket(1, Array[Byte](160.toByte, 0, 1, 2, 3))
    )
  }

  test("SendLossyPacketNotFound") {
    interceptWithTox(ToxFriendCustomPacketException.Code.FRIEND_NOT_FOUND)(
      _.friendSendLossyPacket(1, Array[Byte](200.toByte, 0, 1, 2, 3))
    )
  }

  test("SendLosslessPacketInvalid") {
    interceptWithTox(ToxFriendCustomPacketException.Code.INVALID)(
      _.friendSendLosslessPacket(0, Array[Byte](159.toByte))
    )
  }

  test("SendLossyPacketInvalid") {
    interceptWithTox(ToxFriendCustomPacketException.Code.INVALID)(
      _.friendSendLossyPacket(0, Array[Byte](199.toByte))
    )
  }

  test("SendLosslessPacketEmpty") {
    interceptWithTox(ToxFriendCustomPacketException.Code.EMPTY)(
      _.friendSendLosslessPacket(1, Array[Byte]())
    )
  }

  test("SendLossyPacketEmpty") {
    interceptWithTox(ToxFriendCustomPacketException.Code.EMPTY)(
      _.friendSendLossyPacket(1, Array[Byte]())
    )
  }

  test("SendLosslessPacketNull") {
    interceptWithTox(ToxFriendCustomPacketException.Code.NULL)(
      _.friendSendLosslessPacket(1, null)
    )
  }

  test("SendLossyPacketNull") {
    interceptWithTox(ToxFriendCustomPacketException.Code.NULL)(
      _.friendSendLossyPacket(1, null)
    )
  }

  test("SendLosslessPacketTooLong") {
    interceptWithTox(ToxFriendCustomPacketException.Code.TOO_LONG)(
      _.friendSendLosslessPacket(0, 160.toByte +: Array.ofDim[Byte](ToxCoreConstants.MaxCustomPacketSize))
    )
  }

  test("SendLossyPacketTooLong") {
    interceptWithTox(ToxFriendCustomPacketException.Code.TOO_LONG)(
      _.friendSendLossyPacket(0, 200.toByte +: Array.ofDim[Byte](ToxCoreConstants.MaxCustomPacketSize))
    )
  }

}
